Compilation warnings on macOS

```
[88/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_nc.c.o
../../src/gmt_nc.c:166:199: warning: unused parameter 'n_layers' [-Wunused-parameter]
GMT_LOCAL int gmtnc_io_nc_grid (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *header, unsigned dim[], unsigned origin[], size_t stride, unsigned io_mode, gmt_grdfloat* grid, bool cube, unsigned int n_layers) {
                                                                                                                                                                                                      ^
../../src/gmt_nc.c:943:16: warning: unused variable 'row' [-Wunused-variable]
                unsigned int row, col;
                             ^
../../src/gmt_nc.c:943:21: warning: unused variable 'col' [-Wunused-variable]
                unsigned int row, col;
                                  ^
../../src/gmt_nc.c:2042:32: warning: unused variable 'limit' [-Wunused-variable]
                double level_min, level_max, limit[2];      /* minmax of level variable */
                                             ^
4 warnings generated.
[114/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_api.c.o
../../src/gmt_api.c:5440:32: warning: unused variable 'files' [-Wunused-variable]
        char file[PATH_MAX] = {""}, **files = NULL;
                                      ^
../../src/gmt_api.c:5929:7: warning: unused variable 'file' [-Wunused-variable]
        char file[PATH_MAX] = {""};
             ^
../../src/gmt_api.c:12323:14: warning: '&&' within '||' [-Wlogical-op-parentheses]
                else if (T && strchr (T->arg, '/') || strchr (T->arg, ','))     /* Making more than one output layer */
                         ~~^~~~~~~~~~~~~~~~~~~~~~~ ~~
../../src/gmt_api.c:12323:14: note: place parentheses around the '&&' expression to silence this warning
                else if (T && strchr (T->arg, '/') || strchr (T->arg, ','))     /* Making more than one output layer */
                           ^
                         (                        )
../../src/gmt_api.c:12308:64: warning: unused parameter 'API' [-Wunused-parameter]
GMT_LOCAL char gmtapi_grdinterpolate_type (struct GMTAPI_CTRL *API, struct GMT_OPTION *options) {
                                                               ^
../../src/gmt_api.c:5421:20: warning: unused function 'gmtapi_get_file_count' [-Wunused-function]
GMT_LOCAL uint64_t gmtapi_get_file_count (char **list) {
                   ^
5 warnings generated.
[159/216] Building C object src/CMakeFiles/gmtlib.dir/grdinfo.c.o
../../src/grdinfo.c:723:29: warning: unused variable 'l_min' [-Wunused-variable]
                        uint64_t ij_min, ij_max, l_min, l_max, here = 0, node;
                                                 ^
../../src/grdinfo.c:723:36: warning: unused variable 'l_max' [-Wunused-variable]
                        uint64_t ij_min, ij_max, l_min, l_max, here = 0, node;
                                                        ^
2 warnings generated.
```

--- a/src/gmt_api.c
+++ b/src/gmt_api.c
@@ -5417,14 +5417,6 @@ int GMT_Put_Levels_ (struct GMT_CUBE *C, double *level, uint64_t *n) {
 }
 #endif
 
-
-GMT_LOCAL uint64_t gmtapi_get_file_count (char **list) {
-	/* Determine how many entries before trailing NULL entry */
-	uint64_t k = 0;
-	while (list[k]) k++;
-	return (k);
-}
-
 /*! . */
 GMT_LOCAL struct GMT_CUBE * gmtapi_import_cube (struct GMTAPI_CTRL *API, int object_ID, unsigned int mode, struct GMT_CUBE *cube) {
 	/* Handles the reading of a 3-D data cube given in one of several ways.
@@ -5437,7 +5429,7 @@ GMT_LOCAL struct GMT_CUBE * gmtapi_import_cube (struct GMTAPI_CTRL *API, int obj
 	 * If the cube->data array is NULL it will be allocated for you.
 	 */
 
-	char file[PATH_MAX] = {""}, **files = NULL;
+	char file[PATH_MAX] = {""};
 	int item, new_item, new_ID;
 	bool done = true;
  	uint64_t row, col, kol, row_out, i0, i1, j0, j1, k0, k1, ij, ij_orig;
@@ -5926,7 +5918,6 @@ start_over_import_cube:		/* We may get here if we cannot honor a GMT_IS_REFERENC
 
 /*! Writes out a single cube to destination */
 GMT_LOCAL int gmtapi_export_cube (struct GMTAPI_CTRL *API, int object_ID, unsigned int mode, struct GMT_CUBE *U_obj) {
-	char file[PATH_MAX] = {""};
 	int item, error;
 	bool done = true;
 	unsigned int method;
@@ -12309,6 +12300,7 @@ GMT_LOCAL char gmtapi_grdinterpolate_type (struct GMTAPI_CTRL *API, struct GMT_O
 	unsigned int n_files = 0;
 	char type = 'D';	/* For -S */
 	struct GMT_OPTION *opt = NULL, *E = NULL, *S = NULL, *T = NULL;
+	gmt_M_unused (API);
 
 	for (opt = options; opt; opt = opt->next) {	/* Process all the options given */
 		if (opt->option == GMT_OPT_INFILE) n_files++;
@@ -12320,7 +12312,7 @@ GMT_LOCAL char gmtapi_grdinterpolate_type (struct GMTAPI_CTRL *API, struct GMT_O
 	if (E == NULL && S == NULL) {	/* Interpolating onto a single grid or a new cube */
 		if (T == NULL && n_files > 1)	/* Repackaging multiple grids as a single cube */
 			type = 'U';
-		else if (T && strchr (T->arg, '/') || strchr (T->arg, ','))	/* Making more than one output layer */
+		else if (T && (strchr (T->arg, '/') || strchr (T->arg, ',')))	/* Making more than one output layer */
 			type = 'U';
 		else
 			type = 'G';	/* Making a single grid layer */

--- a/src/gmt_nc.c
+++ b/src/gmt_nc.c
@@ -163,7 +163,7 @@ GMT_LOCAL int gmtnc_n_chunked_rows_in_cache (struct GMT_CTRL *GMT, struct GMT_GR
 }
 
 /* Read and write classic or chunked netcdf files */
-GMT_LOCAL int gmtnc_io_nc_grid (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *header, unsigned dim[], unsigned origin[], size_t stride, unsigned io_mode, gmt_grdfloat* grid, bool cube, unsigned int n_layers) {
+GMT_LOCAL int gmtnc_io_nc_grid (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *header, unsigned dim[], unsigned origin[], size_t stride, unsigned io_mode, gmt_grdfloat* grid, bool cube) {
 	/* io_mode = k_get_netcdf: read a netcdf file to grid
 	 * io_mode = k_put_netcdf: write a grid to netcdf */
 	int status = NC_NOERR;
@@ -940,7 +940,6 @@ GMT_LOCAL int gmtnc_grd_info (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *head
 	}
 	else {	/* Here we are writing */
 		/* Store global attributes */
-		unsigned int row, col;
 		const int *nc_vers = gmtnc_netcdf_libvers();
 		GMT_Report (GMT->parent, GMT_MSG_DEBUG, "netCDF Library version: %d\n", *nc_vers);
 		gmt_M_err_trap (nc_put_att_text (ncid, NC_GLOBAL, "Conventions", strlen(GMT_NC_CONVENTION), GMT_NC_CONVENTION));
@@ -1609,12 +1608,12 @@ int gmt_nc_read_grd (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *header, gmt_g
 
 	/* Read grid */
 	if (dim2[1] == 0)
-		gmtnc_io_nc_grid (GMT, header, dim, origin, HH->stride, k_get_netcdf, pgrid + HH->data_offset, false, 0);
+		gmtnc_io_nc_grid (GMT, header, dim, origin, HH->stride, k_get_netcdf, pgrid + HH->data_offset, false);
 	else {
 		/* Read grid in two parts */
 		unsigned int stride_or_width = HH->stride != 0 ? HH->stride : width;
-		gmtnc_io_nc_grid (GMT, header, dim, origin, stride_or_width, k_get_netcdf, pgrid + HH->data_offset, false, 0);
-		gmtnc_io_nc_grid (GMT, header, dim2, origin2, stride_or_width, k_get_netcdf, pgrid + HH->data_offset + dim[1], false, 0);
+		gmtnc_io_nc_grid (GMT, header, dim, origin, stride_or_width, k_get_netcdf, pgrid + HH->data_offset, false);
+		gmtnc_io_nc_grid (GMT, header, dim2, origin2, stride_or_width, k_get_netcdf, pgrid + HH->data_offset + dim[1], false);
 	}
 
 	/* If we need to shift grid */
@@ -1812,7 +1811,7 @@ int gmt_nc_write_grd (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *header, gmt_
 	/* Write grid */
 	dim[0]    = height,    dim[1]    = width;
 	origin[0] = first_row, origin[1] = first_col;
-	status = gmtnc_io_nc_grid (GMT, header, dim, origin, 0, k_put_netcdf, pgrid, false, 0);
+	status = gmtnc_io_nc_grid (GMT, header, dim, origin, 0, k_put_netcdf, pgrid, false);
 	if (status != NC_NOERR)
 		goto nc_err;
 
@@ -2177,7 +2176,7 @@ int gmt_nc_write_cube (struct GMT_CTRL *GMT, struct GMT_CUBE *C, double wesn[], 
 				gmt_grd_flip_vertical (pgrid, width, height, 0, sizeof(pgrid[0]));
 
 			/* Write grid layer */
-			status = gmtnc_io_nc_grid (GMT, header, dim, origin, 0, k_put_netcdf, pgrid, true, n_layers_used);
+			status = gmtnc_io_nc_grid (GMT, header, dim, origin, 0, k_put_netcdf, pgrid, true);
 			if (status != NC_NOERR)
 				goto nc_err;
 			here += C->header->size;

--- a/src/grdinfo.c
+++ b/src/grdinfo.c
@@ -720,7 +720,7 @@ EXTERN_MSC int GMT_grdinfo (void *V_API, int mode, void *args) {
 		if (Ctrl->T.mode & 2) strncpy (grdfile, opt->arg, PATH_MAX-1);
 
 		if (Ctrl->M.active || Ctrl->L.active) {	/* Must determine the location of global min and max values */
-			uint64_t ij_min, ij_max, l_min, l_max, here = 0, node;
+			uint64_t ij_min, ij_max, here = 0, node;
 			unsigned int col, row;
 			gmt_grdfloat *data = (is_cube) ? U->data : G->data;
 

